specs/interop.md

```diff
@@ -0,0 +1,708 @@
+technically policy and may be changed in the future.
+
+The `unsafe` label has the lowest latency while the `finalized` label has the highest latency.
+A set of properties must be matched before an input or an output can be promoted to the next
+label.
+
+The initiating messages for all dependent executing messages MUST be resolved as safe before an L2 block can transition
+from being unsafe to safe. Users MAY optimistically accept unsafe blocks without any verification of the
+executing messages. They SHOULD optimistically verify the initiating messages exist in destination unsafe blocks
+to more quickly reorganize out invalid blocks.
+
+#### `unsafe` Inputs
+
+- MUST be signed by the p2p sequencer key
+- MAY be reorganized
+- MUST be promoted to a higher level of safety or reorganized out to ensure liveness
+
+`unsafe` inputs are currently gossiped around the p2p network. To prevent denial of service, they MUST
+be signed by the sequencer. This signature represents the sequencer's claim that it
+built a block that conforms to the protocol. `unsafe` blocks exist to give low latency access to the
+latest information. To keep the latency as low as possible, cross chain messages are assumed valid
+at this stage. This means that the remote unsafe inputs are trusted.
+
+An alternative approach to `unsafe` inputs would be to include an SGX proof that the sequencer ran
+particular software when building the block.
+
+#### `cross-unsafe` Inputs
+
+- MUST have valid cross chain messages
+
+`cross-unsafe` represents the `unsafe` blocks that had their cross chain messages fully verified.
+The network can be represented as a graph where each block across all chains are represented as
+a node and then a directed edge between two blocks represents the source block of the initiating
+message and the block that included the executing message.
+
+An input can be promoted from `safe` to `cross-unsafe` when the full dependency graph is resolved
+such that all cross chain messages are verified to be valid and at least one message in the dependency
+graph is still `unsafe`.
+
+#### `safe` Inputs
+
+- MUST be available
+- MAY be reorganized
+- Safe block MUST be invalidated if a reorg occurs
+
+`safe` represents the state in which the `cross-unsafe` dependency graph has been fully resolved
+in a way where all of the data has been published to the data availability layer.
+
+#### `finalized` Inputs
+
+- MUST NOT be reorganized based on Ethereum economic security
+
+`finalized` represents full Proof of Stake economic security on top of the data. This means that
+if the data is reorganized, then validators will be slashed.
+
+## State Transition Function
+
+After the full execution of a block, the set of [logs][log] that are emitted MUST be merklized and
+included in the block header. This commitment MUST be included as the block header's
+[extra data field][block-extra-data]. The events are serialized with using [Simple Serialize][ssz] aka SSZ.
```

Comment:
The yellow paper says extra data is limited to 32 bytes but from the linked execution spec it seems to be arbitrary length. Did it change, or is the execution spec incorrect? If arbitrary length, is the requirement that the commitment is the only data in the extra data field, or can other data be prepended/appended?

Reply:
It must be 32 bytes - https://github.com/ethereum/go-ethereum/blob/8a76a814a2b9e5b4c1a4c6de44cd702536104507/params/protocol_params.go#L31